src/redis.c

```diff
@@ -426,6 +435,12 @@ static int redis_read(void) /* {{{ */
       freeReplyObject(rr);
     }
 
+    if ((rr = redisCommand(rh, "SELECT %d", rn->db)) == NULL) {
+      WARNING("redis plugin: unable to switch to db `%d' on node `%s'.", rn->db,
+              rn->name);
+      goto redis_fail;
+    }
+
     if ((rr = redisCommand(rh, "INFO")) == NULL) {
```

Comment:
I think, `INFO` command shows stats for entire Redis instance, not for selected database, right?
Then new option `Database` better fits to `Query` block. That will be more flexible to use.
What do you think?